### ISSUE TITLE
Serve very long initial page load from 1.0.0.13 to 1.0.1.0

### ISSUE THREAD

**Issue by CookieNick** [nrwl/nx#3941]
## Current Behavior
When upgrading from NX 1.0.0.13 to 1.0.1.0, `nx serve` is very slow. Some basic javascript files take minutes to load on localhost, making it seem it doesn't load at all. When `nx serve` has run and I've waited the full 5 minutes for a specific file to load and then clear cache and reload, it's instant. 

I'm coming from `9.5.1` and tried to upgrade to `1.0.3.1`, everything works except `nx serve`. It builds fine (console output is the same), but opening the page in the browser doesn't work. Building and hosting it in IIS works fine.

## Expected Behavior
As in previous versions, when I run `nx serve`, opening the page in the browser is almost instant.

## Steps to Reproduce
I still need to check if I can create a minimum reproducable case, but that will take a few more days, maybe someone already knows something I might have missed, or how to help debug this, that would be great. 

For me, it was upgrading NX from `9.5.1` to `1.0.3.1. I've isololated it further to be `@nrwl/web` from `1.0.0.13` to `1.0.1.0`. In the meantime, I downgraded that specific package to `1.0.0.13` and everything seems to be working, but it would be best to not be stuck on that version.

I've checked [this comparison](https://github.com/nrwl/nx/compare/10.0.13...10.1.0) to see if anything obvious stands out. I see `copy-webpack-plugin` was upgraded to a new version ( #3514). We use a custom `webpack.config.js` to load some more stuff like fonts, allow for CSS modules and change the default output to include hashes for cache-busting purposes. There's also a change in the code for that with the assets, we have quite a bunch of asset globs which might also be affected (#3328). Though it seems strange as those assets are deployed using `nx build`

### Failure Logs
Here's the screenshot of it loading more than 5 minutes and my custom `web.config.js`.
[Files.zip](https://github.com/nrwl/nx/files/5403979/Files.zip)

### Environment
OS; Windows 10
Node & NPM; 14.14.0 & 6.14.8 (upgraded to latest while debugging)
Notable other frameworks; React and [single-spa](https://single-spa.js.org/)

Not working;
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 10.2.1
  @nrwl/cypress : 10.3.1
  @nrwl/eslint-plugin-nx : 10.3.1
  @nrwl/express : Not Found
  @nrwl/jest : 10.3.1
  @nrwl/linter : 10.3.1
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : 10.3.1
  @nrwl/schematics : Not Found
  @nrwl/tao : 10.2.1
  @nrwl/web : 10.3.1
  @nrwl/workspace : 10.3.1
  typescript : 4.0.3

Working when downgrading @nrwl/web;
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 10.2.1
  @nrwl/cypress : 10.3.1
  @nrwl/eslint-plugin-nx : 10.3.1
  @nrwl/express : Not Found
  @nrwl/jest : 10.3.1
  @nrwl/linter : 10.3.1
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : 10.3.1
  @nrwl/schematics : Not Found
  @nrwl/tao : 10.2.1
  @nrwl/web : 10.0.13
  @nrwl/workspace : 10.3.1
  typescript : 4.0.3

**Comment by mandarini** [comment]
Hello there!

Thank you for filing an issue! I looked at the screenshot attached, and, yes, 6 minutes is a long time to load! 

I tried to reproduce your issue locally, and I could not do it :(. I understand that indeed your issue may have to do with your specific use case, custom configs, extra assets, fonts, etc. It would be very helpful if you could provide a minimum reproduction repository (I understand that you may not be able to provide your actual codebase and assets, but at least a repo with some placeholders that would showcase your issue). 

Also, can you please adjust the title of the issue to be a bit more clear? I understand that you started with version `9.5.1` and then you used `nx migrate` to migrate to `10.3.1`? Is my understanding correct?

And then, as you say, you isolated the issue when upgrading the `@nrwl/web` package from `10.0.13` to `10.1.0`?

So, you upgraded from `9.5.1` to `10.3.1`, and then started downgrading `@nrwl/web` to spot the point where it "broke"? So, only for the `@nrwl/web` package, you went to `10.0.13` and it worked (normal loading time),  and then you went to `10.1.0` and the issue reappeared (long loading time)?

**Comment by CookieNick** [comment]
Hi!

Thanks for getting back to me. I was working on a reproduceable case, but work got in the way and the workaround was fine for a few weeks. There's a lot of stuff coming together which makes it a little time consuming, as I'm working with a lot of customized stuff too. I'll look into it again soon. 

You completely understand my steps, that's great. As you've described is exactly what I've done, though at first I was up/downgrading all of NX and in the end just the `@nrwl/web` package. I tried to make the title as clear as I could, but I'm happy to take suggestions to make it clearer for you and the rest of the team!

**Comment by mandarini** [comment]
Thank you @CookieNick ! I will wait until you have the reproducible case available, and then I will try my best to solve your issue! :)

**Comment by CookieNick** [comment]
While stripping out all business sensitive stuff, I got somewhat further in isolating this issue and I think it's in the way globs are handled. In `workspace.json`, on the project I'm building, in the asset array, I have an input glob like this;
```json
{
    "input": "../external-content/modules", 
    "glob": "*/dist/*",
     "output": "modules"
}
```

In `external-content/modules`, we have a few external modules we still need to migrate to migrate to NX, for now they are built externally and downloaded. We only need the `dist` folder in that folder, but the folder structure is like;

* modules
  * moduleA (there are around 10 modules)
    * dist
    * node_modules
    * public
    * src
    .gitignore
   package.json
  ...etc

Whilst mocking this sctructure, it worked like expected. But when I use the production folder structure, it's slow like described. I found out that if I remove the `node_modules` folder in my production folder, it does work.

Therefore, I suspect there's something in the glob that wants to parse the entire `node_modules` folder which takes ages. That's something I expect when the glob would be `**/dist/*` as described in [https://github.com/isaacs/node-glob/blob/master/README.md](node-glob) for example, that's why I put it as `*/dist/*`.

I hope this is enough to find out some solution to fix the glob, as sending the project with the `node_modules` folder would be a bit much. I'll zip it anyway if it's needed, but I'd have to send it through WeTransfer or something else you'd prefer.
